--- compiler.js	2012-03-13 17:03:06.000000000 +0200
+++ new_compiler.js	2014-12-13 06:48:24.000000000 +0200
@@ -18,6 +18,9 @@
   // remove whitespace according to Mustache spec
   var rIsWhitespace = /\S/,
       rQuot = /\"/g,
+	  sQuot = /\'/g,
+	  rScriptOpen = /<script/g,
+	  rScriptClose = /<\/script/g,
       rNewline =  /\n/g,
       rCr = /\r/g,
       rSlash = /\\/g,
@@ -245,12 +248,17 @@
   function esc(s) {
     return s.replace(rSlash, '\\\\')
             .replace(rQuot, '\\\"')
+			.replace(sQuot, '\\\'')
+			.replace(rScriptOpen, '<\\x73cript')
+			.replace(rScriptClose, '</\\x73cript')
+
             .replace(rNewline, '\\n')
             .replace(rCr, '\\r');
   }
 
   function chooseMethod(s) {
-    return (~s.indexOf('.')) ? 'd' : 'f';
+    if (~s.indexOf('('))  return 'fc'; //zx patch
+    return    (~s.indexOf('.')) ? 'd' : 'f';
   }
 
   function walk(tree) {
