--- marked.js	2014-03-10 07:59:41.000000000 +0200
+++ new_marked.js	2014-12-13 15:16:42.000000000 +0200
@@ -20,7 +20,7 @@
   lheading: /^([^\n]+)\n *(=|-){2,} *(?:\n+|$)/,
   blockquote: /^( *>[^\n]+(\n(?!def)[^\n]+)*\n*)+/,
   list: /^( *)(bull) [\s\S]+?(?:hr|def|\n{2,}(?! )(?!\1bull )\n*|\s*$)/,
-  html: /^ *(?:comment|closed|closing) *(?:\n{2,}|\s*$)/,
+  html: /^ *(?:zxtag|comment|closed|closing) *(?:\n{2,}|\s*$)/,
   def: /^ *\[([^\]]+)\]: *<?([^\s>]+)>?(?: +["(]([^\n]+)[")])? *(?:\n+|$)/,
   table: noop,
   paragraph: /^((?:[^\n]+\n?(?!hr|heading|lheading|blockquote|tag|def))+)\n*/,
@@ -50,6 +50,7 @@
 
 block.html = replace(block.html)
   ('comment', /<!--[\s\S]*?-->/)
+  ('zxtag', /<#[\s\S]*?>/)
   ('closed', /<(tag)[\s\S]+?<\/\1>/)
   ('closing', /<tag(?:"[^"]*"|'[^']*'|[^'">])*?>/)
   (/tag/g, block._tag)
@@ -359,7 +360,7 @@
         type: this.options.sanitize
           ? 'paragraph'
           : 'html',
-        pre: cap[1] === 'pre' || cap[1] === 'script' || cap[1] === 'style',
+        pre: cap[1] === 'pre' || cap[1] === 'script' || cap[1] === 'style' || cap[1] === undefined, //zxx
         text: cap[0]
       });
       continue;
@@ -437,7 +438,7 @@
         Error('Infinite loop on byte: ' + src.charCodeAt(0));
     }
   }
-
+  
   return this.tokens;
 };
 
